### PR TITLE
refactor how button group elements are styled

### PR DESCRIPTION
**Pull request by radnan**

### CODE REVIEW

File: apps/storybook/src/forms/ButtonGroup.stories.tsx [hunk 1]
```diff
@@ -1,33 +1,112 @@
...
+  render: (args) => (
+    <Flex alignItems="center">
+      {presets.map(([preset, label]) => (
+        <ButtonGroup {...args} key={preset}>
+          <BasicContent preset={preset}>{label}</BasicContent>
+        </ButtonGroup>
+      ))}
+      <ButtonGroup {...args}>
+        <BasicContent colorScheme="danger" variant="ghost">
+          Danger Secondary
```
Comment (MunifTanjim) [inline apps/storybook/src/forms/ButtonGroup.stories.tsx:86]: Danger Ghost?

File: apps/storybook/src/forms/ButtonGroup.stories.tsx [hunk 2]
```diff
@@ -1,33 +1,112 @@
...
+
+const presets = [
+  ["default", "Default"],
+  ["primary", "Primary"],
+  ["danger", "Danger"],
+  ["danger-outline", "Outline"],
+  ["secondary", "Secondary"],
+] as const;
+
+export const Presets: Story = {
```
Comment (MunifTanjim) [inline apps/storybook/src/forms/ButtonGroup.stories.tsx:76]: Should ghost buttons have border between them?

image

Comment (radnan) [reply apps/storybook/src/forms/ButtonGroup.stories.tsx]: still debating that with designers

File: apps/storybook/src/forms/ButtonGroup.stories.tsx (hunk 1 above)
Comment (radnan) [reply apps/storybook/src/forms/ButtonGroup.stories.tsx:86]: addressed

**Pull request merged**